db/src/schema.rs

```diff
@@ -245,6 +245,7 @@ impl SchemaTypeChecking for Schema {
                 // Ref coerces a little: we interpret some things depending on the schema as a Ref.
                 (&ValueType::Ref, TypedValue::Long(x)) => Ok(TypedValue::Ref(x)),
                 (&ValueType::Ref, TypedValue::Keyword(ref x)) => self.require_entid(&x).map(|entid| TypedValue::Ref(entid)),
+                (&ValueType::Instant, TypedValue::Instant(x)) => Ok(TypedValue::Instant(x)),
```

Comment:
Also, this commit seems to have other stuff in it, not related to `:db.type/inst`.